### PR TITLE
Add import functionality and table to visualise the data

### PR DESCRIPTION
**Pull request by Yuenmin**
The PR includes the following changes:

- Allow user to import dataset by uploading CSV file or specifying URL
- Persist the imported data in IndexedDB
- Table to visualize the imported data with pagination and searching
- Allow user to add, edit and delete a row after importing the dataset
- Allow user to batch delete rows by selecting checkboxes
- Allow user to remove the imported dataset
- Developers can get the data imported from the IndexedDB with pagination, sorting and filtering by using Dexie useLiveQuery() hook

### CODE REVIEW

**Comment by omar-al-hendi** [comment]
Amazing work @Yuenmin, it is really nice to see how you dived into carbon components. 

However, I have few Suggestions will contribute to the readability and maintainability of the project.

File: src/pages/ImportPage.tsx [hunk 1]
```diff
@@ -0,0 +1,756 @@
...
+
+  useEffect(() => {
+    setIsLoading(true);
+    const fetchData = async () => {
+      await loadData();
+      setIsLoading(false);
+    };
+    fetchData();
+  }, []);
+
```
Comment (omar-al-hendi) [inline src/pages/ImportPage.tsx:74]: The loader feature from React Router can be used to git rid of this code which will reduce the amount of logic that you write.

File: src/pages/ImportPage.tsx [hunk 2]
```diff
@@ -0,0 +1,756 @@
...
+    }
+    const fields = Object.keys(data[0]).filter((field) => field !== "id");
+    setHeaders(
+      fields.map((field) => ({
+        header: field,
+        key: field,
+      }))
+    );
+    setRows(data.map((t) => ({ ...t, id: String(t.id) })));
+  };
```
Comment (omar-al-hendi) [inline src/pages/ImportPage.tsx:117]: This will be the body of the loader function

File: src/pages/ImportPage.tsx [hunk 3]
```diff
@@ -0,0 +1,756 @@
...
+          Cancel
+        </Button>
+        <Button kind="danger" onClick={deleteRow}>
+          Delete
+        </Button>
+      </ModalFooter>
+    </ComposedModal>
+  );
+};
+
```
Comment (omar-al-hendi) [inline src/pages/ImportPage.tsx:755]: The Modals share a lot of the logic, What do you think of creating a one single modal component, that can handle the logic of all the modal ? @Yuenmin

Comment (omar-al-hendi) [inline src/pages/ImportPage.tsx:755]: In this page we have a lot of requests from our application to the DB, I highly recommend using a library like React Query that can help us write a better fetching and mutation logic, feel free to check it [here ](https://tanstack.com/query/v3/).  It also helps us handle the error and the loading status.

File: src/pages/ImportPage.tsx [hunk 4]
```diff
@@ -0,0 +1,756 @@
...
+  const [headers, setHeaders] = useState<Array<DataTableHeader>>([]);
+  const [page, setPage] = useState<number>(1);
+  const [pageSize, setPageSize] = useState<number>(10);
+  const [open, setOpen] = useState<ModalType>("");
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [targetId, setTargetId] = useState<number | number[]>(NaN);
+  const [editData, setEditData] = useState<any>({});
+  const [showNotification, setShowNotification] = useState<
+    "success" | "error" | ""
+  >("");
```
Comment (omar-al-hendi) [inline src/pages/ImportPage.tsx:64]: If we have decided to go with React Query and the loaders feature, the rest of the logic can be written here using useReducer which help us handle such cases more efficiently

File: src/pages/ImportPage.tsx [hunk 5]
```diff
@@ -0,0 +1,756 @@
...
+        <ToastNotification
+          kind="error"
+          role="status"
+          timeout={4000}
+          title="An error occurred"
+          onClose={closeNotification}
+        />
+      ) : (
+        <div style={{ height: 50 }}></div>
+      )}
```
Comment (omar-al-hendi) [inline src/pages/ImportPage.tsx:187]: The code can be written in this way, 

```
{showNotification !== "" &&  (
        <ToastNotification
          kind={showNotification}
          role="status"
          timeout={4000}
          title={showNotification === "sucess" ? "Success" : "an Error Occured" }
          onClose={closeNotification}
        />
      )}
      
```

**Review by omar-al-hendi — CHANGES_REQUESTED** [review]
I have also tried the adding row, editing row features, 
It did not work for me, I think because the data does not get mutated.

**Comment by omar-al-hendi** [comment]
I also recommend splitting your page into multiple components to help us track bugs in the feature and reduce the amount of maintainability (SOC), an example might be.

- Pages 
  -   Import
     -  Data Table
     - Modal 
     -  (Other Components)
     - Page (page logic)

**Review by sulaimanmhashim — DISMISSED** [review]

File: src/pages/ImportPage.tsx (hunk 1 above)
Comment (Yuenmin) [reply src/pages/ImportPage.tsx:74]: I think it is not suitable to use loader here as I would like to control when to refresh the data and when to show the other components that are related to the data refresh (e.g. loading overlay, popup notification)

File: src/pages/ImportPage.tsx (hunk 3 above)
Comment (Yuenmin) [reply src/pages/ImportPage.tsx]: The change is included in the latest commit.

Comment (Yuenmin) [reply src/pages/ImportPage.tsx]: I used Dexie, an indexedDB wrapper library to speed up our development. The developers can now use the hooks provided by Dexie to make the query. References: [https://dexie.org/docs/libs/[dexie-react-hooks](https://dexie.org/docs/libs/dexie-react-hooks)](url) [https://dexie.org/docs/API-Reference](url)

File: src/pages/ImportPage.tsx (hunk 5 above)
Comment (Yuenmin) [reply src/pages/ImportPage.tsx]: This change has been included in the latest commit.

**Review by omar-al-hendi — APPROVED** [review]

**Review by OoiJSen — DISMISSED** [review]

**Review by CXiann — APPROVED** [review]

**Pull request merged**